Valigator.AspNetCore/AspNetCore/VerifiableExtensions.cs

```diff
@@ -39,7 +39,7 @@ public static class VerifiableExtensions
 					some => some
 						.Match
 						(
-							o => dataContainer.WithValue(data, Option.Some((TValue)o)), 
+							o => o is TDataValue ? data.WithUncheckedValue((TDataValue)o) : dataContainer.WithValue(data, Option.Some((TValue)o)), 
```

Comment:
I don't think this is right. I think this whole thing isn't going to handle set/unset/null correctly.

I think this is all you need:
```value
	.Match
	(
		isSet => isSet
			.Match
			(
				notNull => notNull is TValue castValue ? dataContainer.WithValue(data, Option.Some(castValue)) : throw new ValidateAttributeDoesNotSupportTypeException(notNull.GetType(), typeof(TValue)),
				() => dataContainer.WithNull(data)
			) 
		() => data
	)
	.Verify()
	.TryGetValue()
	.Select(o => (object)o);```

Reply:
For my purposes (the `ValidateModelBinderAttribute`) I don't think these extensions need to change at all. I have removed `IVerifiable` from the attribute because it no longer implements `IValidateType<TDataValue>`.

I will look into this a bit for `IValidateAttribute` because I think that is the only situation we care about this in now.

Reply:
Johannes is going to deal with this. (not in scope of my changes ;-) )